models/schemas/levelSchema.ts

```diff
@@ -57,4 +62,45 @@ const LevelSchema = new mongoose.Schema<Level>({
   },
 });
 
+LevelSchema.index({ slug: 1 }, { name: 'slug_index'});
+
+LevelSchema.pre('save', function(next) {
+  // update slug if name changed
+  if (this.isModified('name')) {
+    UserModel.findById(this.userId).then(user => {
+      this.slug = user.name + '/' + this.name.replace(/\s+/g, '-').toLowerCase();
+      next();
+    });
+  } else {
+    next();
+  }
+});
+
+// Now do updateOne
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const onUpdateCheck = async function(me:any, next:any) {
+  if (me.getUpdate().$set.name) {
+    console.log(me._conditions._id);
+    LevelModel.findById(me._conditions._id).then(level => {
+      UserModel.findById(level.userId).then(user => {
+        me.getUpdate().$set.slug = user.name + '/' + me.getUpdate().$set.name.replace(/\s+/g, '-').toLowerCase();
```

Comment:
What happens if someone updates their username? Or if this is run every time we retrieve a level then there is no issue.

Also could maybe do LevelModel.findById().populate(userId.. ) instead of two separate queries?

Reply:
Yes you can currently update your username/email here: https://pathology.k2xl.com/account

Reply:
updated